### PR TITLE
Feature/ts

### PR DESCRIPTION
**Pull request by Vylpes**

### CODE REVIEW

**Comment by Vylpes** [comment]
Need to fix events, I'm unable right now to pass variables into the execution method since you can't have variables be different depending on the event if you use an overwrite method.

**Comment by Vylpes** [comment]
Should hopefully be fixed now

File: .env.template [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Security Warning! Do not commit this file to any VCS!
```
Comment (VylpesTester) [inline .env.template:1]: Document this file

File: src/client/client.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    public start() {
+        if (!process.env.BOT_TOKEN) throw "BOT_TOKEN is not defined in .env";
+        if (!process.env.BOT_PREFIX) throw "BOT_PREFIX is not defined in .env";
+        if (!process.env.FOLDERS_COMMANDS) throw "FOLDERS_COMMANDS is not defined in .env";
+        if (!process.env.FOLDERS_EVENTS) throw "FOLDERS_EVENTS is not defined in .env";
+
+        super.on("message", this._events.onMessage);
+        super.on("ready", this._events.onReady);
```
Comment (VylpesTester) [inline src/client/client.ts:26]: This should also be an event in the event class

File: src/client/events.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+                    prefix: prefix,
+                    name: name,
+                    args: args,
+                    message: message,
+                },
+            };
+        }
+
+        return {
+            valid: true,
```
Comment (VylpesTester) [inline src/client/events.ts]: Should be false?

File: src/client/util.ts [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        message: Message;
+    }
+}
+
+// Util Class
+export class Util {
+    public loadCommand(name: string, args: string[], message: Message): IUtilResponse {
+        if (!message.member) return {
+            valid: false,
+            message: "Member is not part of member",
```
Comment (VylpesTester) [inline src/client/util.ts]: "Member is not part of message"

**Review by VylpesTester — CHANGES_REQUESTED** [review]

**Review by VylpesTester — APPROVED** [review]

**Pull request merged**